### PR TITLE
docs: Dataplane performance impact

### PR DESCRIPTION
**Pull request by boruszak**
### Description
In a Slack conversation, Matt Keeler responded to a question about changes to performance when using Consul Dataplane, in light of the architectural difference when not using client agents. 

This update adds a section to the Dataplane index page to summarize these impacts.

This PR also updates the diagram on the page to highlight the relationship between the orchestrator and Consul to provide additional clarity.

### Links
[Original Slack convo](https://hashicorp.slack.com/archives/CPEPBFDEJ/p1667850333077589)

### PR Checklist

* [ ] updated test coverage
* [X] external facing docs updated
* [X] not a security concern

### CODE REVIEW

File: website/content/docs/connect/dataplane/index.mdx [hunk 1]
```diff
@@ -17,7 +17,18 @@ In standard deployments, Consul uses a control plane that contains both _server
 
 Consul Dataplane manages Envoy proxies and leaves responsibility for other functions to the orchestrator. As a result, it removes the need to run client agents on every pod. In addition, services no longer need to be reregistered to a local client agent after restarting a service instance, as a client agent’s lack of access to persistent data storage in Kubernetes deployments is no longer an issue.
 
-![Diagram of Consul Dataplanes in Kubernetes deployment](/img/dataplanes-diagram.png)
```
Comment (jkirschner-hashicorp) [inline website/content/docs/connect/dataplane/index.mdx:20]: Is there an old img file that is now unused/should be removed?

File: website/content/docs/connect/dataplane/index.mdx [hunk 2]
```diff
@@ -17,7 +17,18 @@ In standard deployments, Consul uses a control plane that contains both _server
...
+![Diagram of Consul Dataplanes in Kubernetes deployment](/img/k8s-dataplanes-architecture.png)
+
+### Impact on performance
+
+The most significant differences between traditional deployments and Consul Dataplane deployments result from the removal of node-level client agents with gossip communication. They are replaced by _dataplanes_, which are the sidecars injected alongside each service instance that handle communication between Consul servers and Envoy proxies.
+
+Be aware of the following changes and their impact on your network's performance:
+
+1. Consul servers consume additional resources in order to generate xDS resources for Envoy proxies. In our internal load tests, performing at high scale and churn resulted in additional CPU utilization rates under 10%.
```
Comment (david-yu) [inline website/content/docs/connect/dataplane/index.mdx:28]:
```suggestion
1. Consul servers consume additional resources in order to generate xDS resources for Envoy proxies. In our internal load tests, performing at high scale and churn resulted in additional CPU utilization rates under 10% on the control plane. 
```

**Review by david-yu — APPROVED** [review]
Approved, added a suggestion to make it clearer where the 10% utilization is occurring.

File: website/content/docs/connect/dataplane/index.mdx [hunk 3]
```diff
@@ -17,7 +17,18 @@ In standard deployments, Consul uses a control plane that contains both _server
...
+
+### Impact on performance
+
+The most significant differences between traditional deployments and Consul Dataplane deployments result from the removal of node-level client agents with gossip communication. They are replaced by _dataplanes_, which are the sidecars injected alongside each service instance that handle communication between Consul servers and Envoy proxies.
+
+Be aware of the following changes and their impact on your network's performance:
+
+1. Consul servers consume additional resources in order to generate xDS resources for Envoy proxies. In our internal load tests, performing at high scale and churn resulted in additional CPU utilization rates under 10% on the control plane. 
+1. As you deploy more services, the resource usage for dataplanes grows on a linear scale.
+1. Envoy reconfigurations are rate limited to prevent excessive configuration changes from generating significant load on the servers.
```
Comment (riddhi89) [inline website/content/docs/connect/dataplane/index.mdx:30]: I think one more point worth surfacing that helps with optimizing server performance is proxy configuration load balancing. 
So another point could be "Proxy configuration load-balancing has been implemented on Consul servers to help with even spread of load across the cluster of servers pro-actively".

**Pull request merged**